sphinx/ext/coverage.py

```diff
@@ -121,6 +124,14 @@ def write_c_coverage(self) -> None:
                 write_header(op, filename)
                 for typ, name in sorted(undoc):
                     op.write(' * %-50s [%9s]\n' % (name, typ))
+                    if self.config.coverage_print_missing_c_items:
+                        if self.app.quiet or self.app.warningiserror:
+                            logger.warning(__('undocumented c api: %s [%s] in file %s'),
+                                           name, typ, filename)
```

Comment:
The warning level log is not good for this case (including linkcheck builder...). With `-W` option, sphinx-build fails ASAP if any warnings found. But it would be better to show full error on CI process for me.

Reply:
Ah, I'd forgotten about `--keep-going` option. It helps `-W` on the CI server. Please forget my comment above.

Reply:
I run sphinx-build two times:
1. with `-q` to get all warnings
2. with `-QW` to get fail/success status.

I copied this form somewhere else. But maybe the `-keep-going` option can reduce this to one run.